src/sagas/storage.ts

```diff
@@ -0,0 +1,242 @@
+import { SagaIterator } from 'redux-saga';
```

Comment:
The functions in this files are way too big! Can consider splitting them up and externalising the URLs to constants! Originally copied this code from @blackening when he hacked something together! I think this class can be refactored a little!

Reply:
I made some small changes. I went through other saga files and found they also had comparatively big functions so I guess it is ok!